fifo.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"context"
+	"encoding/csv"
+	"fmt"
+	"io"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/sirupsen/logrus"
+)
+
+type FifoHandler struct {
+	pipe   io.Reader
+	events <-chan fsnotify.Event
+
+	running map[string]context.CancelFunc
+
+	handleNotification notificationHandlerFunc
+}
+
+type notificationHandlerFunc func(ctx context.Context, notification Notification)
+
+func NewFifoHandler(cfg notifyConfig, pipe io.Reader, events <-chan fsnotify.Event) (*FifoHandler, error) {
+
+	p, err := cfg.NewProvider()
+	if err != nil {
+		return nil, err
+	}
+	fh := &FifoHandler{
+		pipe:               pipe,
+		events:             events,
+		running:            map[string]context.CancelFunc{},
+		handleNotification: defaultNotificatonHandler(p, cfg),
+	}
+	return fh, nil
+}
+
+func (h FifoHandler) HandleFifo(ctx context.Context) error {
+	// Yes, this can actually be parsed as a CSV file with spaces as separators and it handles quoted string the same way a shell does.
+	r := csv.NewReader(h.pipe)
+	r.Comma = ' '
+
+	for {
+		select {
+		case e := <-h.events:
+			switch e.Op {
+			case fsnotify.Write:
+				lines, err := r.ReadAll()
+				if err != nil {
+					logrus.Errorf("Failed to read from fifo: %s", err)
+					continue
```

Comment:
I guess that's fine since we wait for the next update to try again. So no log spamming/looping should occur.